x-pack/plugins/observability_solution/slo/public/pages/slos/components/common/quick_filters.tsx

```diff
@@ -27,13 +27,13 @@ export function QuickFilters({
   initialState: { tagsFilter, statusFilter },
   onStateChange,
 }: Props) {
-  const [controlGroupAPI, setControlGroupAPI] = useState<AwaitingControlGroupAPI>();
+  const [controlGroupAPI, setControlGroupAPI] = useState<ControlGroupRendererApi | undefined>();
 
   useEffect(() => {
     if (!controlGroupAPI) {
       return;
     }
-    const subscription = controlGroupAPI.onFiltersPublished$.subscribe((newFilters) => {
+    const subscription = controlGroupAPI.filters$.pipe(skip(1)).subscribe((newFilters = []) => {
```

Comment:
why do we skip(1)?

Reply:
`filters$` outputs as soon as the control group loads, regardless of if there are selections, whereas `onFiltersPublished$` did not publish when there were no selections. I didn't dig into it too ya'lls code too deeply, but this change was causing an infinite loop because, whenever `onStateChange` is called, its reference changes and the `useEffect` is triggered. The `skip` was the quickest and easiest solution that didn't change functionality :bow: